kroxylicious-operator/src/main/java/io/kroxylicious/kubernetes/operator/ProxyConfigSecret.java

```diff
@@ -43,33 +62,56 @@ static String secretName(KafkaProxy primary) {
-                                adminHttp:
-                                  endpoints:
-                                    prometheus: {}
-                                virtualClusters:
-                                  demo:
-                                    targetCluster:
-                                      bootstrap_servers: TARGET_BOOTSTRAP_SERVERS
-                                    clusterNetworkAddressConfigProvider:
-                                      type: PortPerBrokerClusterNetworkAddressConfigProvider
-                                      config:
-                                        bootstrapAddress: localhost:9292
-                                        brokerAddressPattern: BROKER_ADDRESS_PATTERN
-                                    logNetwork: false
-                                    logFrames: false
-                                """
-                                .replace(
-                                        "TARGET_BOOTSTRAP_SERVERS", primary.getSpec().getBootstrapServers())
-                                .replace("BROKER_ADDRESS_PATTERN", ProxyService.serviceName(primary))))
+                .withStringData(Map.of(CONFIG_YAML_KEY, generateProxyConfig(primary)))
                 .build();
     }
+
+    String generateProxyConfig(KafkaProxy primary) {
+
+        List<Clusters> clusters = ClustersUtil.distinctClusters(primary);
+        var virtualClusters = clusters.stream()
+                .collect(Collectors.toMap(
+                        Clusters::getName,
+                        cluster -> getVirtualCluster(primary, cluster),
+                        (v1, v2) -> v1, // Dupes handled below
+                        LinkedHashMap::new));
+
+        Configuration configuration = new Configuration(
+                new AdminHttpConfiguration(null, null, new EndpointsConfiguration(new PrometheusMetricsConfig())),
+                virtualClusters,
+                List.of(),
+                List.of(),
+                false);
+
+        ObjectMapper mapper = ConfigParser.createObjectMapper();
+        try {
+            return mapper.writeValueAsString(configuration);
+        }
+        catch (JsonProcessingException e) {
+            throw new UncheckedIOException(e);
+        }
+    }
+
+    private static VirtualCluster getVirtualCluster(KafkaProxy primary, Clusters cluster) {
+        var o = ClustersUtil.distinctClusters(primary);
+        var clusterNum = o.indexOf(cluster);
+        return new VirtualCluster(
+                new TargetCluster(cluster.getUpstream().getBootstrapServers(), Optional.empty()),
+                new ClusterNetworkAddressConfigProviderDefinition(
+                        "PortPerBrokerClusterNetworkAddressConfigProvider",
+                        new PortPerBrokerClusterNetworkAddressConfigProvider.PortPerBrokerClusterNetworkAddressConfigProviderConfig(
+                                new HostPort("localhost", 9292 + (100 * clusterNum)),
```

Comment:
A limit of ~560 clusters, yeah I guess thats probably safe. 

It does however emphasise that we still have a lot of work todo to sensibly expose clusters

Reply:
Yeah, this is definitely a problem. There's too many hard-coded constants floating around. But I'd like to merge this and then start focussing more on the Kube networking orchestration. It doesn't seem to be simple for any of the options. 
* For port-per-broker we need to know the number of brokers in the cluster in order to know how many ports to in the cluster's `Service`. 
* But for SNI we _also_ need to know in order to have the Services in order to get the DNS names.
* Those are problem problems which apply to a single cluster. Once we we've solved those we can start to think more about how to divvy up the port number space between clusters for the port-per-broker case.

Reply:
I should have been more explicit. I agree nothing to do in scope of this PR.